en/docs/install-and-setup/setup/security/logins-and-passwords/harshicrop-vault-extension.md

```diff
@@ -0,0 +1,298 @@
+    keystore.identity.store.secretProvider=org.wso2.carbon.securevault.DefaultSecretCallbackHandler
+    keystore.identity.key.password=identity.key.password
+    keystore.identity.key.secretProvider=org.wso2.carbon.securevault.DefaultSecretCallbackHandler
+    carbon.secretProvider=org.wso2.securevault.secret.handler.SecretManagerSecretCallbackHandler
+
+    secVault.enabled=true
+    secretRepositories=vault
+    secretRepositories.vault.provider=org.wso2.carbon.securevault.hashicorp.repository.HashiCorpSecretRepositoryProvider
+    secretRepositories.vault.properties.address=https://127.0.0.1:8200
+    secretRepositories.vault.properties.namespace=ns1
+    secretRepositories.vault.properties.enginePath=wso2apim
+    secretRepositories.vault.properties.engineVersion=2
+    secretRepositories.vault.properties.authType=ROOT_TOKEN
+    ```
+
+    !!! note
+        In production, you should always use the vault address with TLS enabled.
+
+5. Add the following lines to the `<APIM_HOME>/repository/conf/log4j2.properties` file:
+
+    ```properties
+    logger.org-wso2-carbon-securevault-hashicorp.name=org.wso2.carbon.securevault.hashicorp
+    logger.org-wso2-carbon-securevault-hashicorp.level=INFO
+    logger.org-wso2-carbon-securevault-hashicorp.additivity=false
+    logger.org-wso2-carbon-securevault-hashicorp.appenderRefCARBON_CONSOLE. ref = CARBON_CONSOLE
+    ```
+
+6. Then append `org-wso2-carbon-securevault-hashicorp` to the `loggers` list in the same file as follows:
+
+    ```properties
+    loggers = AUDIT_LOG, trace-messages, ..., org-wso2-carbon-securevault-hashicorp
+    ```
+
+### Step 3: Update passwords with their aliases
+
+1. Open the `deployment.toml` file in the `<APIM_HOME>/repository/conf/` directory and add
+   the `[secrets]` configuration section **at the bottom of the file** as shown below.
+   Give an alias for the passwords and put the value as blank (`""`).
+
+    ```toml
+    [secrets]
+    admin_password = ""
+    ```
+
+2. Add the encrypted password alias to the relevant sections in the `deployment.toml`
+   file by using a placeholder: `$secret{alias}`. For example:
+
+    ```toml
+    [super_admin]
+    username="admin"
+    password="$secret{admin_password}"
+    ```
+
+### Step 4: Start the server
+
+1. Provide the `VAULT_TOKEN` to the prompted message in the console or by creating a new file in the `<APIM_HOME>` 
+   directory. The file should be named according to your Operating System.
+
+    === "Linux"
+        ```
+        The file name should be `hashicorpRootToken-tmp`.
+        ```
+      
+    === "Windows"
+        ```
+        The file name should be `hashicorpRootToken-tmp.txt`.
+        ```
+
+    !!! note
+        When you add "tmp" to the file name, note that this will automatically get deleted from the file system after the server starts. Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
```

Comment:
```suggestion
        When you add `tmp` to the file name, note that this will automatically get deleted from the file system after the server starts. Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
```

Reply:
corrected in 1d8bdffa01d44e5da79703dd2060f0a621bb659a